lib/improver/generate_ancillaries/generate_ancillary.py

```diff
@@ -0,0 +1,249 @@
+
+    Raises
+    -------
+    IOError: if input ancillary cannot be found in StaGE directory provided
+    """
+    stage_ancil = glob(stage_ancil)
+    if len(stage_ancil) > 0:
+        standard_ancil = iris.load(stage_ancil[0])[0]
+    else:
+        msg = 'Cannot find input ancillary. Tried StaGE: {}'
+        raise IOError(msg.format(stage_ancil))
+    return standard_ancil
+
+
+class CorrectLandSeaMask(object):
+    """
+    Round landsea mask to binary values
+
+    Corrects interpolated land sea masks to boolean values of
+    False [sea] and True [land].
+    """
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string"""
+        result = ('<CorrectLandSeaMask>')
+        return result
+
+    @staticmethod
+    def process(standard_landmask):
+        """Read in the interpolated landmask and assign values round
+            values < 0.5 to False and >=0.5 to True.
+
+        Parameters
+        ----------
+        standard_landmask: cube
+            input landmask on standard grid.
+
+        Returns
+        -------
+        standard_landmask : cube
+            output landmask of boolean values.
+        """
+        mask_sea = np.ma.masked_less(standard_landmask.data, 0.5).mask
+        standard_landmask.data[mask_sea] = False
+        mask_land = np.ma.masked_greater(standard_landmask.data, 0.).mask
+        standard_landmask.data[mask_land] = True
+        return standard_landmask
+
+
+class GenerateOrographyBandAncils(object):
+    """
+    Generate topographic band ancillaries for the standard grids.
+
+    Reads StaGE output orography files first, then generates binary mask
+    of land points within the orography band specified.
+    """
+    def __init__(self):
+        pass
+
+    def __repr__(self):
+        """Represent the configured plugin instance as a string."""
+        result = ('<GenerateOrographyBandAncils>')
+        return result
+
+    @staticmethod
+    def gen_orography_masks(
+            standard_orography, standard_landmask, key, thresholds):
+        """
+        Function to generate topographical band masks.
+
+        Parameters
+        -----------
+        standard_orography : cube
+            The standard orography, read from StaGE.
+        standard_landmask : cube
+            The landmask generated by gen_landmask.
+        key : string
+            Key from THRESHOLD_DICT which descibes type of topography band.
+        thresholds: float or list
+            Upper and/or lower thresholds of the current topographical band.
+
+        Returns
+        -------
+        mask_cube : cube
+            Cube containing topographical band mask.
+
+        Raises
+        ------
+        KeyError: if the key does not match any in THRESHOLD_DICT.
+        """
+        coords = standard_orography.coords()
+        if key == 'max_land_threshold':  # mask everything above max bound
+            orog_band = np.ma.masked_greater(
```

Comment:
Is there a particular reason to use `np.ma.masked_greater` here? The same result could be achieved with:
```python
1. orog_band = (standard_orography.data > thresholds[0]).astype(int)
2. orog_band = np.greater(standard_orography.data, thresholds[0]).astype(int)
```
As far as understand it, this section of code is doing the following:
1. A binary array is created, if a threshold is exceeded or not.
```python
[[0 0 0]
 [0 1 0]
 [0 1 0]]
```
2. Sea values, i.e. where the land-sea mask is zero, are masked.
```python
[[0 -- --]
 [0 -- --]
 [0 1 0]]
```
3. The masked array is filled with a fill value, so that the sea points all have the same fill value.

If this is correct, I think that it would be worth spelling out what the aim is, potentially with a simple example in the docstring.